### ISSUE TITLE
[BUG] Upgrade 1.2.0 -> 1.2.1 is stuck in “Waiting for VM live-migration or shutdown...(1 left)” even though there is NO VM running

### ISSUE THREAD

**Issue by Martin-Weiss** [harvester/harvester#4725]
**Describe the bug**
During the upgrade of Harvester 1.2.0 to 1.2.1 the upgrade gets stuck with the message "`Waiting for VM live-migration or shutdown...(1 left)`"

But no VM is running at all:
![grafik](https://github.com/harvester/harvester/assets/13362573/b8850acf-0ac4-4ffa-a1db-2646c8c2229f)

Status of the VM:
```
status:
  conditions:
    - lastProbeTime: '2023-09-15T09:54:53Z'
      lastTransitionTime: '2023-09-15T09:54:53Z'
      message: virt-launcher pod is terminating
      reason: PodTerminating
      status: 'False'
      type: Ready
    - lastProbeTime: null
      lastTransitionTime: null
      status: 'True'
      type: LiveMigratable
  created: true
  printableStatus: Stopped
  volumeSnapshotStatuses:
    - enabled: false
      name: cdrom-disk
      reason: 2 matching VolumeSnapshotClasses for longhorn-image-dzr67
    - enabled: false
      name: rootdisk
      reason: 2 matching VolumeSnapshotClasses for longhorn
    - enabled: false
      name: virtio-container-disk
      reason: >-
        Snapshot is not supported for this volumeSource type
        [virtio-container-disk]
    - enabled: false
      name: cloudinitdisk
      reason: Snapshot is not supported for this volumeSource type [cloudinitdisk]
```

**To Reproduce**
Upgrade a 3 node harvester cluster from 1.2.0 to 1.2.1 with all VMs stopped while one VM has a similar status as above.

**Expected behavior**
Upgrade does not get stuck based on "not running VMs"

**Environment**
 - Harvester ISO version: 1.2.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal

**Comment by Martin-Weiss** [comment]
[1.20 - 1.12 hvst-upgrade-xdt69-upgradelog-archive-2023-11-13T13-10-19Z.zip](https://github.com/harvester/harvester/files/13346383/1.20.-.1.12.hvst-upgrade-xdt69-upgradelog-archive-2023-11-13T13-10-19Z.zip)

**Comment by w13915984028** [comment]
Per discussion and investigation, below differences are related:

https://docs.harvesterhci.io/v1.1/troubleshooting/vm#a-vm-stopped-using-the-harvester-ui

https://docs.harvesterhci.io/v1.1/troubleshooting/vm#a-vm-stopped-using-the-vms-poweroff-command

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.3.0-rc1` (24/01/25).

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Can upgrade from v1.2.1 to v1.3.0-rc1 when all the VMs shutdown from the OS $~~$</summary>

  1. When we login to the windows 2022 and opensuse leap 15.4 and shutdown machine from system UI (not from Harvester dashboard
  
 ![image](https://github.com/harvester/harvester/assets/29251855/09ac1ca8-98fb-4491-b696-5c78c4be3ec4)


  2. We can successfully upgrade to from v1.2.1 to v1.3.0-rc1 and remain the OS state, did not encounter stuck in Waiting form VM live-migration..

https://github.com/harvester/harvester/assets/29251855/2456f6fb-59f7-41ee-bc13-4647d4c10043



 </details>



### Test Information
* Test Environment: Three nodes harvester on baremetal machines
* Harvester version: `v1.3.0-rc1` (24/01/25).

### Verify Steps
<details><summary>Upgrade with VM shutdown from OS </summary>

  1. Prepare 3 nodes v1.2.1 Harvester cluster
  2. Create a Windows server 2022 VM
  3. Create a openSUSE leap 15.4 VM
  4. Login to Windows VM and shutdown machine from menu
  5. Login to openSUSE VM , use command to shutdown machine
  6. Upgrade Harvester to v1.3.0-rc1
  7. Check the upgrade process

</details>
